From 7b5990a96ddd0408d244cfcb83dd66aa658ab809 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.liu@freescale.com>
Date: Tue, 11 Sep 2012 18:16:12 +0800
Subject: [PATCH 2083/2508] ENGR00223797-1 ARM:IPUv3:Add bypass reset in pdata

This patch adds bypass reset field support in ipuv3
platform data, so that a platform may choose not to
reset ipuv3 when doing probe, which may support
smooth transition from bootloader splashimage to
system UI.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
(cherry picked from commit e558b2b6877a2b230c564a1c621bbba3d546f5f7)
---
 arch/arm/plat-mxc/include/mach/ipu-v3.h |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/arch/arm/plat-mxc/include/mach/ipu-v3.h b/arch/arm/plat-mxc/include/mach/ipu-v3.h
index 82da102..48daf4d 100755
--- a/arch/arm/plat-mxc/include/mach/ipu-v3.h
+++ b/arch/arm/plat-mxc/include/mach/ipu-v3.h
@@ -745,6 +745,13 @@ struct imx_ipuv3_platform_data {
 	void (*pg) (int);
 
 	char *csi_clk[2];
+
+	/*
+	 * Bypass reset to avoid display channel being
+	 * stopped by probe since it may starts to work
+	 * in bootloader.
+	 */
+	bool bypass_reset;
 };
 
 #endif /* __MACH_IPU_V3_H_ */
-- 
1.7.7.4

